CMake/FindLZ4.cmake

```diff
@@ -0,0 +1,15 @@
+find_path(LZ4_INCLUDE_DIR lz4.h)
```

Comment:
Is this file needed at all? I would expect modern CMake to be able to find the system lz4 by itself, and our externals should be taken care of with the `dolphin_find_optional_system_library()` call.

Reply:
Yes I can remove this and will do so on my next rebase.

Reply:
Okay I tested this and yeah it's not needed. I also noticed a few other CMake issues, this should fix all of them: https://github.com/AdmiralCurtiss/dolphin/commit/1539fa378136b8987cf710f8e6187e4daeaa22ed